* <grobian@gentoo.org>: small compilation fix for Darwin, and produce
                        correct(ly named) dylibs

--- libopenjpeg/opj_malloc.h
+++ libopenjpeg/opj_malloc.h
@@ -75,7 +75,7 @@
 #else /* Not WIN32 */
 	#if defined(__sun)
 			#define HAVE_MEMALIGN
-		#elif defined(__GNUC__)
+		#elif defined(__GNUC__) && !defined(__APPLE__)
 			#if defined(__FreeBSD__)
 			#include <stdlib.h>
 			#include <osreldate.h>
--- Makefile
+++ Makefile
@@ -26,8 +26,15 @@
 
 TARGET  = openjpeg
 STATICLIB = lib$(TARGET).a
+ifeq ($(TARGOS),Darwin)
+SHAREDLIB = lib$(TARGET).$(VER_MAJOR).$(VER_MINOR).dylib
+LIBNAME = lib$(TARGET).$(VER_MAJOR).dylib
+LIBCMD = -dynamiclib -install_name $(INSTALL_LIBDIR)/$(LIBNAME)
+else
 SHAREDLIB = lib$(TARGET).so.$(VER_MAJOR).$(VER_MINOR)
 LIBNAME = lib$(TARGET).so.$(VER_MAJOR)
+LIBCMD = -shared -Wl,-soname,$(LIBNAME)
+endif
 
 
 
@@ -54,7 +61,7 @@
 	$(AR) r $@ $(MODULES)
 
 $(SHAREDLIB): $(MODULES)
-	$(CC) $(LDFLAGS) -shared -Wl,-soname,$(LIBNAME) -o $@ $(MODULES) $(LIBRARIES)
+	$(CC) $(LDFLAGS) $(LIBCMD) -o $@ $(MODULES) $(LIBRARIES)
 
 install: OpenJPEG
 	install -d '$(DESTDIR)$(INSTALL_LIBDIR)' '$(DESTDIR)$(INSTALL_INCLUDE)'
